crypto/x509/x509type.c

```diff
@@ -41,6 +41,9 @@ int X509_certificate_type(const X509 *x, const EVP_PKEY *pkey)
     case EVP_PKEY_EC:
         ret = EVP_PK_EC | EVP_PKT_SIGN | EVP_PKT_EXCH;
         break;
+    case EVP_PKEY_ED25519:
+        ret = EVP_PKT_SIGN;
```

Comment:
This still looks odd to me. So we've changed the NID_X25519 -> EVP_PKEY_ED25519, but should the return flags see a similar change? e.g. `ret = EVP_PK_ED25519 | EVP_PKT_SIGN`